### ISSUE TITLE
Only part of the rows are inside the table when rows are added to the table

### ISSUE THREAD

**Issue by harri-rajala-koivu-solutions** [OfficeDev/office-js#3191]
Only part of the rows are inside the table when rows are added to the table, and other rows are outside. When it happens depends on how many rows have values and how many rows have an empty string.

## Your Environment
* Platform [PC desktop, Mac, iOS, Office on the web]: PC desktop
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: 2301 (Build 16026.20200 Click-to-Run)
* Operating System: Windows 11 Pro, 10.0.22621 Build 22621
 
## Expected behavior
Excel should always add all rows to the table.

## Current behavior
Only part of the rows are inside the table when rows are added to the table, and other rows are outside. When it happens depends on how many rows have values and how many rows have an empty string.

For example, you have 3100 rows, the first 1500 and the last **10 rows** have a value, and the other rows have an empty value. In this case, the table is broken because it only has 1500 rows, and it should have 3100 rows.

On the other hand, if you have 3100 rows, the first 1500 and the last **11 rows** have a value, and the other rows have an empty value. In this case, the table is ok and has all 3100 rows. 

Also, with 1501 and 10 rows table is ok. It has 3100 rows.

You can reproduce the issue with the following code. You can also try to change valuesStart and valuesEnd parameters.

## Steps to reproduce
```typescript
await Excel.run(async (context: Excel.RequestContext) => {
    const sheet = context.workbook.worksheets.getActiveWorksheet();
    const addTable = async (column: string, rowCount: number, valuesStart: number, valuesEnd: number) => {
        const table = sheet.tables.add(`${column}2:${column}2`, true);
        table.getHeaderRowRange().values = [[`${rowCount} row, ${valuesStart} - ${valuesEnd}`]];
        const values = [];
        for (let i = 1; i <= rowCount; i++) {
            if (i <= valuesStart || i > (rowCount - valuesEnd)) {
                values.push([`X ${String(i).padStart(4, "0")}`]);
            } else {
                values.push([""]);
            }
        }
        table.getHeaderRowRange().getCell(0, 0).getRowsBelow(values.length).values = values;
        table.getRange().format.autofitColumns();
        await context.sync();
    };
    await addTable("B", 3010, 1500, 9);
    await addTable("D", 3010, 1500, 10);
    await addTable("F", 3010, 1501, 10);
    await addTable("H", 3010, 1500, 11);
    await addTable("J", 3010, 1500, 12);
});
```

## Context
Our Excel add-in does not work because of this bug.

**Comment by harri-rajala-koivu-solutions** [comment]
Hi @penglongzhaochina. This ticket is assigned to you. 
Can you please comment on this issue? 
When is this critical bug fixed?

**Comment by penglongzhaochina** [comment]
Hi @harri-rajala-koivu-solutions Sorry for late, Our engineer is under investigation. I will contact you once we have some progress. Thanks.

**Comment by dingjin-ms** [comment]
Hi @harri-rajala-koivu-solutions , thank you for reporting the issue and sorry for any inconvenience. I could definitely repro the problem and open a bug #7663840 internally to track the problem. 

But while we are investigating the fix, please allow me to offer you a workaround to see if we could unblock you first.

Could you please try using `table.rows.add` rather than ranges.values to add rows to the table?
```
await Excel.run(async (context: Excel.RequestContext) => {
    const sheet = context.workbook.worksheets.getActiveWorksheet();
    const addTable = async (column: string, rowCount: number, valuesStart: number, valuesEnd: number) => {
      const values = [];
      for (let i = 1; i <= rowCount; i++) {
        if (i <= valuesStart || i > rowCount - valuesEnd) {
          values.push([`X ${String(i).padStart(4, "0")}`]);
        } else {
          values.push([""]);
        }
      }

      const table = sheet.tables.add(`${column}2:${column}${2}`, true);
      table.getHeaderRowRange().values = [[`${rowCount} row, ${valuesStart} - ${valuesEnd}`]];
      table.rows.add(0, values, false); // you could set alwaysInsert option to false if you expect no needed values below the table, this option will perf better as it will only change ranges.values but not expanding the row.
      await context.sync();
    };

    await addTable("B", 3010, 1500, 9);
    await addTable("D", 3010, 1500, 10);
    await addTable("F", 3010, 1501, 10);
    await addTable("H", 3010, 1500, 11);
    await addTable("J", 3010, 1500, 12);
  });
```

**Comment by harri-rajala-koivu-solutions** [comment]
@dingjin-ms Thank you for the workaround. I will check if we can use that.
Please check another ticket, https://github.com/OfficeDev/office-js/issues/3013, that I created last year and is still open. Are these two bugs related?

**Comment by dingjin-ms** [comment]
@harri-rajala-koivu-solutions I believe so, we will take these 2 issues into account together to see if we could get some fix.

**Comment by ruijiaMS** [comment]
This issue has the same root cause with #3013. please leverage the same comments and solutions on that. Thanks!

**Comment by ruijiaMS** [comment]
closed as duplicate
